src/hb-ot-name-table.hh

```diff
@@ -116,6 +116,29 @@ struct NameRecord
     return (p == 0 ||
             (p == 3 && (e == 0 || e == 1 || e == 10)));
   }
+  
+  static int cmp (const void *pa, const void *pb)
+  {
+    const NameRecord *a = (const NameRecord *)pa;
+    const NameRecord *b = (const NameRecord *)pb;
+
+    if (a->platformID != b->platformID)
+      return a->platformID < b->platformID ? -1 : +1;
```

Comment:
A slightly more compact way to do this is

```
return a->platformID - b->platformID;
```

Reply:
Also put the `cmp` in `NameRecord` itself.

Reply:
> A slightly more compact way to do this is
> return a->platformID - b->platformID;

This isn't the common pattern on the table, will apply it later.

> Also put the cmp in NameRecord itself.

Won't work as is, needs the following also, will provide it as a separate PR.

```diff
diff --git a/src/hb-array.hh b/src/hb-array.hh
index cbd6485a..2c1bb2fd 100644
--- a/src/hb-array.hh
+++ b/src/hb-array.hh
@@ -145,16 +145,22 @@ struct hb_array_t : hb_iter_with_fallback_t<hb_array_t<Type>, Type&>
     return not_found;
   }
 
-  hb_sorted_array_t<Type> qsort (int (*cmp_)(const void*, const void*))
+  HB_INTERNAL static int qsort_cmp (const void *pa, const void *pb)
+  {
+    Type *a = (Type *) pa;
+    Type *b = (Type *) pb;
+    return b->cmp (*a);
+  }
+  hb_sorted_array_t<Type> qsort ()
   {
     if (likely (length))
-      hb_qsort (arrayZ, length, this->get_item_size (), cmp_);
+      hb_qsort (arrayZ, length, this->get_item_size (), qsort_cmp);
     return hb_sorted_array_t<Type> (*this);
   }
-  hb_sorted_array_t<Type> qsort ()
+  hb_sorted_array_t<Type> qsort (int (*cmp_)(const void*, const void*))
   {
     if (likely (length))
-      hb_qsort (arrayZ, length, this->get_item_size (), Type::cmp);
+      hb_qsort (arrayZ, length, this->get_item_size (), cmp_);
     return hb_sorted_array_t<Type> (*this);
   }
   void qsort (unsigned int start, unsigned int end)
diff --git a/src/hb-ot-name-table.hh b/src/hb-ot-name-table.hh
index 0da56718..b3b1a9e3 100644
--- a/src/hb-ot-name-table.hh
+++ b/src/hb-ot-name-table.hh
@@ -112,30 +112,27 @@ struct NameRecord
   {
     unsigned int p = platformID;
     unsigned int e = encodingID;
-    
+
     return (p == 0 ||
             (p == 3 && (e == 0 || e == 1 || e == 10)));
   }
-  
-  static int cmp (const void *pa, const void *pb)
-  {
-    const NameRecord *a = (const NameRecord *)pa;
-    const NameRecord *b = (const NameRecord *)pb;
 
-    if (a->platformID != b->platformID)
-      return a->platformID < b->platformID ? -1 : +1;
+  int cmp (const NameRecord &record)
+  {
+    if (record.platformID != platformID)
+      return record.platformID < platformID ? -1 : +1;
 
-    if (a->encodingID != b->encodingID)
-      return a->encodingID < b->encodingID ? -1 : +1;
+    if (record.encodingID != encodingID)
+      return record.encodingID < encodingID ? -1 : +1;
 
-    if (a->languageID != b->languageID)
-      return a->languageID < b->languageID ? -1 : +1;
+    if (record.languageID != languageID)
+      return record.languageID < languageID ? -1 : +1;
 
-    if (a->nameID != b->nameID)
-      return a->nameID < b->nameID ? -1 : +1;
+    if (record.nameID != nameID)
+      return record.nameID < nameID ? -1 : +1;
 
-    if (a->length != b->length)
-      return a->length < b->length ? -1 : +1;
+    if (record.length != length)
+      return record.length < length ? -1 : +1;
 
     return 0;
   }
```